### PR TITLE
Add California Privacy Laws Notice of Collection to Card Form

### PR DESCRIPTION
**Pull request by jaxdesmarais**
### Summary of changes

 - Add California Privacy Laws Notice of Collection to Card Form

| Initial Card Form | Details Card Form |
|-----|-----|
|Simulator Screenshot - iPhone 14 - 2023-07-26 at 08 33 48|Simulator Screenshot - iPhone 14 - 2023-07-26 at 08 33 56|

 ### Checklist

 - [x] Added a changelog entry

### Authors

- @scannillo @jaxdesmarais

### CODE REVIEW

File: Sources/BraintreeDropIn/BTCardFormViewController.m [hunk 1]
```diff
@@ -285,6 +297,11 @@ - (void)updateRequiredFields {
     }
 }
 
+- (void) tappedPrivacyPolicy: (UITapGestureRecognizer *) gestureRecognizer {
+    NSURL *privacyPolicy = [NSURL URLWithString:@"https://www.paypal.com/us/legalhub/home?locale.x=en_US"];
+    [[UIApplication sharedApplication] openURL:privacyPolicy options:@{} completionHandler:nil];
```
Comment (jaxdesmarais) [inline Sources/BraintreeDropIn/BTCardFormViewController.m:302]: We can also consider opening this an an `ASWebAuthenticationSession` if we want to keep this in context

File: Sources/BraintreeDropIn/BTCardFormViewController.m [hunk 2]
```diff
@@ -285,6 +297,11 @@ - (void)updateRequiredFields {
     }
 }
 
+- (void) tappedPrivacyPolicy: (UITapGestureRecognizer *) gestureRecognizer {
+    NSURL *privacyPolicy = [NSURL URLWithString:@"https://www.paypal.com/us/legalhub/home?locale.x=en_US"];
```
Comment (scannillo) [inline Sources/BraintreeDropIn/BTCardFormViewController.m:301]: Same comment as I had on the web PR [here](https://github.com/braintree/braintree-web-drop-in/pull/899/files#r1275042361)

**Comment by scannillo** [comment]
Looks good to me! I tested with large font size and everything scrolls and fits OK. Also checked that the screen reader indeed reads off the privacy label as a "button" and not just text.

File: Sources/BraintreeDropIn/BTCardFormViewController.m (hunk 2 above)
Comment (jaxdesmarais) [reply Sources/BraintreeDropIn/BTCardFormViewController.m]: Updated to remove the locale, good call: https://github.com/braintree/braintree-ios-drop-in/pull/420/commits/619c4ae1500d89ae598b4d687037004ee994b98c

File: Sources/BraintreeDropIn/BTCardFormViewController.m (hunk 1 above)
Comment (scannillo) [reply Sources/BraintreeDropIn/BTCardFormViewController.m:302]: I think ASWeb would be better for conversion. But we should keep it consistent across SDKs. Android would have to use browser switch a well.

**Review by sshropshire — APPROVED** [review]

**Review by scannillo — APPROVED** [review]

**Pull request merged**